sql/src/main/java/org/apache/apex/malhar/sql/planner/RelNodeVisitor.java

```diff
@@ -0,0 +1,457 @@
+    for (RelNode input : n.getInputs()) {
+      inputStreams.add(traverse(input));
+    }
+
+    RelInfo currentNodeRelInfo;
+
+    if (n instanceof Aggregate) {
+      currentNodeRelInfo = visitAggregate((Aggregate)n, inputStreams);
+    } else if (n instanceof Calc) {
+      currentNodeRelInfo = visitCalc((Calc)n, inputStreams);
+    } else if (n instanceof Collect) {
+      currentNodeRelInfo = visitCollect((Collect)n, inputStreams);
+    } else if (n instanceof Correlate) {
+      currentNodeRelInfo = visitCorrelate((Correlate)n, inputStreams);
+    } else if (n instanceof Delta) {
+      currentNodeRelInfo = visitDelta((Delta)n, inputStreams);
+    } else if (n instanceof Exchange) {
+      currentNodeRelInfo = visitExchange((Exchange)n, inputStreams);
+    } else if (n instanceof Project) {
+      currentNodeRelInfo = visitProject((Project)n, inputStreams);
+    } else if (n instanceof Filter) {
+      currentNodeRelInfo = visitFilter((Filter)n, inputStreams);
+    } else if (n instanceof Sample) {
+      currentNodeRelInfo = visitSample((Sample)n, inputStreams);
+    } else if (n instanceof Sort) {
+      currentNodeRelInfo = visitSort((Sort)n, inputStreams);
+    } else if (n instanceof TableModify) {
+      currentNodeRelInfo = visitTableModify((TableModify)n, inputStreams);
+    } else if (n instanceof TableScan) {
+      currentNodeRelInfo = visitTableScan((TableScan)n, inputStreams);
+    } else if (n instanceof Uncollect) {
+      currentNodeRelInfo = visitUncollect((Uncollect)n, inputStreams);
+    } else if (n instanceof Window) {
+      currentNodeRelInfo = visitWindow((Window)n, inputStreams);
+    } else if (n instanceof Join) {
+      currentNodeRelInfo = visitJoin((Join)n, inputStreams);
+    } else {
+      currentNodeRelInfo = defaultValue(n, inputStreams);
+    }
+
+    if (currentNodeRelInfo != null && inputStreams.size() != 0) {
+      for (int i = 0; i < inputStreams.size(); i++) {
+        RelInfo inputStream = inputStreams.get(i);
+        Operator.OutputPort outputPort = inputStream.getOutPort();
+        Operator.InputPort inputPort = currentNodeRelInfo.getInputPorts().get(i);
+
+        String streamName = OperatorUtils.getUniqueStreamName(inputStream.getRelName(),
+            currentNodeRelInfo.getRelName());
+        Class schema;
+        if (inputStream.getOutRelDataType() != null) {
+          schema = getSchemaForRelDataType(streamName, inputStream.getOutRelDataType());
+        } else if (inputStream.getClazz() != null) {
+          schema = inputStream.getClazz();
+        } else {
+          throw new RuntimeException("Unexpected condition reached.");
+        }
+        dag.setOutputPortAttribute(outputPort, Context.PortContext.TUPLE_CLASS, schema);
+        dag.setInputPortAttribute(inputPort, Context.PortContext.TUPLE_CLASS, schema);
+        dag.addStream(streamName, outputPort, inputPort);
+      }
+    }
+
+    if (currentNodeRelInfo.getOutPort() == null) {
```

Comment:
As such there is no plan for outputting to 2 different tables. In relation algebraic representation of SQL, there might be multiple input points but a single output point. There is a query I've put similar to this which will verify the possibility of that: https://lists.apache.org/thread.html/4989caa200ea49b5bad1a64f059c44f1a13d2e98a43e297cbeaf6a82@%3Cdev.calcite.apache.org%3E